File: Contest_Solutions/Code-e/02_Spring_2021_Contests/Contest_15/C++/Maximum_Subarray_Easy_Version/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main(){
	int t;
	cin>>t;
	while(t--){
	    int n;
	    cin>>n;
	    int a[n];
	    for(int i = 0; i < n; i++)
	        cin>>a[i];
	    int ans = -1e8, sum = 0;
	    for(int i = 0; i < n; i++){
	        sum = max(sum+a[i], a[i]);
	        ans = max(ans, sum);
	    }
        cout<<ans<<endl;
        
	}
	return 0;
}

File: Contest_Solutions/Code-e/02_Spring_2021_Contests/Contest_15/Java/Maximum_Subarray_Easy_Version/main.java
import java.util.*;

public class main{
	public static void main( String[] args ){
		Scanner cin = new Scanner(System.in);
		int t = cin.nextInt();
		while(t > 0){
		    int n = cin.nextInt();
		    int[] a = new int[n];
		    for(int i = 0; i < n; i++)
		        a[i] = cin.nextInt();
		    int ans = (int)-1e8, sum = 0;
		    for(int i = 0; i < n; i++){
		        sum = Math.max(sum + a[i], a[i]);
		        ans = Math.max(ans, sum);
		    }
		    System.out.println(ans);
		    t--;
		}
	}
}